presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -913,10 +918,11 @@ private Scope legacyVisitQuerySpecification(QuerySpecification node, Optional<Sc
             // to pass down to analyzeFrom
 
             Scope sourceScope = analyzeFrom(node, scope);
+            Scope sourceScopeWithWindow = createWindowScope(node, Optional.of(sourceScope));
```

Comment:
If you believe that outer scope WINDOW definition should not leak into the subquery's scope, then it's not what you have in this PR, IIUC. Especially with this implementation: https://github.com/prestodb/presto/pull/8307/commits/d3aa447ad85ac757e1016628e9fd484b7937fa38#diff-48ecf80fcd7db9168b3eb678246addecR201.
However, I think how it's implemented is actually correct because we should be able to pull WINDOW definition from outer scope but giving preference to inner scope definitions and take outer ones only in case we lack inner's.